StoryMap/StoryMap/Screens/Map/Coordinator/MapCoordinator.swift

```diff
@@ -30,6 +34,34 @@ class MapCoordinator: CoordinatorType {
     
     private func showAddStory() {
         addStoryCoordinator = AddStoryCoordinator()
+        addStoryCoordinator?.onShowStory = { [weak self] story in
+            self?.showStoryDetail(with: story)
+        }
         addStoryCoordinator?.start(presenter.topViewController)
     }
+    
+    private func showStoryDetail(with story: Story) {
+        storyDetailCoordinator = StoryDetailCoordinator(story: story)
+        storyDetailCoordinator?.presenter = presenter
+        storyDetailCoordinator?.start(nil)
+    }
+
+    private func setupTestStory() {
+        let store = RealmDataProvider.shared
+        let title: String? = "Lulu Waterfall"
+        let image = StyleKit.image.make(
+            from: StyleKit.image.examples.waterfall,
+            with: .alwaysTemplate
+        )
+
+        let imageData = image?.jpegData(compressionQuality: 1)
+        guard let title = title, let imageData = imageData else {
+            return
+        }
+
+        let story = Story(title: title, image: imageData)
+        store.write(object: story)
```

Comment:
I know it's only for testing, but keep in mind that this story will be saved every time the app is opened. However, when you're passing the story as an argument, it doesn't really need to be in database at all.